### PR TITLE
Setting Rust's logging level to `RADIXDLT_LOG_LEVEL` env

### PR DESCRIPTION
**Pull request by jakrawcz-rdx**
This applies the **Quick Fix** described at https://radixdlt.atlassian.net/browse/NODE-576?focusedCommentId=19240.

A more comprehensive fix (i.e. "rust->java log bridge" mentioned [here](https://radixdlt.atlassian.net/browse/NODE-576?focusedCommentId=19238)) will come in another PR.

This PR does NOT address any aspects of log formatting/coloring.

### What has changed in the logs?

When `RADIXDLT_LOG_LEVEL=warn ./gradlew --info build runAllIntegrationTests >out.txt 2>err.txt`

**Before this PR:**

Then `out.txt` contains `>=WARN` lines from Java and ALL lines from Rust:

```
[0K[1mcom.radixdlt.integration.targeted.capabilities.NodeCapabilityTests[m
[0K[1m  Test [22mensure_disabled_messages_are_discarded[32m PASSED[31m (13.9s)[m

com.radixdlt.integration.targeted.capabilities.NodeCapabilityTests > ensure_capabilities_are_sent_in_the_handshake_message STANDARD_OUT
    [1;33m2024-01-08T14:19:05,282 [WARN/StandardHostIp/Test worker] - Example warn from Java.[m
[2m2024-01-08T13:19:05.575403Z[0m [32m INFO[0m [2mstate_manager::store::rocks_db[0m[2m:[0m Example info from Rust
[2m2024-01-08T13:19:05.576202Z[0m [33m WARN[0m [2mstate_manager::store::rocks_db[0m[2m:[0m Example warn from Rust
```

**After this PR:**

Then `out.txt` contains `>=WARN` lines from Java and `>=WARN` lines from Rust:

```
[0K[1mcom.radixdlt.integration.targeted.capabilities.NodeCapabilityTests[m
[0K[1m  Test [22mensure_disabled_messages_are_discarded[32m PASSED[31m (13.9s)[m

com.radixdlt.integration.targeted.capabilities.NodeCapabilityTests > ensure_capabilities_are_sent_in_the_handshake_message STANDARD_OUT
    [1;33m2024-01-08T14:19:05,282 [WARN/StandardHostIp/Test worker] - Example warn from Java.[m
[2m2024-01-08T13:19:05.576202Z[0m [33m WARN[0m [2mstate_manager::store::rocks_db[0m[2m:[0m Example warn from Rust
```

⚠️ **What still has not changed and can only be changed by the more comprehensive solution mentioned earlier?**

When `RADIXDLT_LOG_LEVEL=info ./gradlew build runAllIntegrationTests >out.txt 2>err.txt` (i.e. no `--info` == should not print any application's output)

Then `out.txt` contains NO lines from Java and `>=INFO` (i.e. default `RADIXDLT_LOG_LEVEL`) lines from Rust:

```
[0K[1mcom.radixdlt.integration.targeted.capabilities.NodeCapabilityTests[m
[0K[1m  Test [22mensure_disabled_messages_are_discarded[32m PASSED[31m (13.9s)[m

com.radixdlt.integration.targeted.capabilities.NodeCapabilityTests > ensure_capabilities_are_sent_in_the_handshake_message STANDARD_OUT
[2m2024-01-08T13:19:05.575403Z[0m [32m INFO[0m [2mstate_manager::store::rocks_db[0m[2m:[0m Example info from Rust
[2m2024-01-08T13:19:05.576202Z[0m [33m WARN[0m [2mstate_manager::store::rocks_db[0m[2m:[0m Example warn from Rust
```

### CODE REVIEW

File: core-rust/state-manager/src/state_computer.rs [hunk 1]
```diff
@@ -97,18 +97,6 @@ use std::ops::Deref;
 use std::sync::Arc;
 use std::time::{Instant, SystemTime};
 
-#[derive(Debug, Categorize, Encode, Decode, Clone, Default)]
-pub struct LoggingConfig {
-    pub engine_trace: bool,
-    pub state_manager_config: StateComputerLoggingConfig,
-}
-
-// TODO: Replace this with better loglevel integration
```
Comment (jakrawcz-rdx) [inline core-rust/state-manager/src/state_computer.rs:106]: _(note from author)_

I believe "better loglevel integration" is exactly what I do in this PR; so I now simply log the rejected transaction at DEBUG (instead of using this flag).

File: core-rust/state-manager/src/store/jmt_gc.rs [hunk 2]
```diff
@@ -271,11 +271,11 @@ mod tests {
     #[test]
     fn iterates_substates_from_deleted_partition_in_dfs_order() {
         // Arrange: we need a valid tree store state - easiest to just use JMT's infra
-        let mut tree_store = TypedInMemoryTreeStore::new();
+        let tree_store = TypedInMemoryTreeStore::new();
```
Comment (jakrawcz-rdx) [inline core-rust/state-manager/src/store/jmt_gc.rs:274]: _(note from author)_

This is a drive-by, just because of warnings.

**Review by siy — APPROVED** [review]
LGTM
